File: javascript/group_anagram_str.js
const group_anagrams = (strings)=>{
    //write your code here
    finalArr = []
    dictFinal = {}
    for (i of strings){
        console.log(i)
        x = i.split("").sort().join("")
        console.log(x)
        if (x in dictFinal)
            dictFinal[x].push(i)
        else
        dictFinal[x] = [i]
    }
    for (i in dictFinal)
        finalArr.push(dictFinal[i])

    return finalArr
}


console.log(group_anagrams(['abc', 'cat', 'cab', 'atc']))

File: python/group_anagram_str.py
def group_anagrams(strings):
    #write your code here
    finalArr = []
    dictFinal = {}
    for i in range(len(strings)):
        print(strings[i])
        x=sorted(strings[i])
        x= ''.join(x)
        print(x)
        if x in dictFinal:
            dictFinal[x].append(strings[i])
        else:
            dictFinal[x] = [strings[i]]
    for i in dictFinal:
        finalArr.append(dictFinal[i])
     
    return finalArr
      
    
print(group_anagrams(['abc','cat', 'cab', 'atc']))